developer_guides/apps/components/component-api.rst

```diff
@@ -6,28 +6,24 @@ Component API
 Component Device "Constructor"
 ******************************
 
-Called to create a new component device::
+The following call creates a new component device::
 
    struct comp_dev *(*new)(struct sof_ipc_comp *comp);
 
-The framework calls ``comp_ops::new()`` to create a new instance of the
-component, so called component device. All required data object should be
+This framework calls ``comp_ops::new()`` to create a new instance of the
+component, called the component device. All required data objects should be
 allocated from the run-time heap (``RZONE_RUNTIME``).
 
-Note that any component specific private data is allocated separately and
-pointer to that one is connected to the common ``comp_dev`` structure's
-`private` field by calling ``comp_set_drvdata()`` function. There is
-complimentary ``comp_get_drvdata()`` available in order to retrieve the private
-data structure in other component routines.
+Note that any component-specific private data is allocated separately and the pointer to that one is connected to the common ``comp_dev`` structure's
```

Comment:
@mmaka1 I am not clear on the use of "pointer" here. Is it a verb or a noun? We can talk privately if you like.